rs/sns/governance/proto/ic_sns_governance/pb/v1/governance.proto

```diff
@@ -2153,6 +2155,56 @@ message AdvanceTargetVersionRequest {
 // The response to a request to advance the target version of the SNS.
 message AdvanceTargetVersionResponse {}
 
+// Represents a single entry in the upgrade journal.
+message UpgradeJournalEntry {
+  oneof event {
+    UpgradeStepsRefreshed upgrade_steps_refreshed = 1;
+    TargetVersionSet target_version_set = 2;
+    UpgradeStarted upgrade_started = 3;
+    UpgradeCompleted upgrade_completed = 4;
+  }
+  optional uint64 timestamp_seconds = 5;
+
+  message UpgradeStepsRefreshed {
+    optional Governance.Versions upgrade_steps = 2;
+  }
+
+  message TargetVersionSet {
+    optional Governance.Version version = 1;
+  }
+
+  message UpgradeStarted {
+    optional Governance.Version current_version = 1;
+    optional Governance.Version expected_version = 2;
+    oneof triggerer {
+      ProposalId upgrade_sns_to_next_version = 3;
+      Empty behind_target_version = 4;
+    }
+  }
+
+  message UpgradeCompleted {
+    optional string human_readable = 1;
+
+    oneof status {
+      Empty succeeded = 2;
+      Empty timed_out = 3;
```

Comment:
Consider adding a field to indicate what the timeout was.

Reply:
Not sure this is the right approach - let's revisit